pkg/repo/space_store.go

```diff
@@ -0,0 +1,232 @@
+package repo
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/multiformats/go-multibase"
+	"github.com/storacha/go-ucanto/core/delegation"
+	"github.com/storacha/go-ucanto/did"
+	"github.com/storacha/go-ucanto/principal/ed25519/signer"
+	"github.com/storacha/go-ucanto/ucan"
+
+	"github.com/storacha/guppy/pkg/didmailto"
+)
+
+// TODO since we are persisting private keys here, might be worth using a datastore
+// instead of a json file
+type Space struct {
+	DID         did.DID   `json:"did"`
+	Name        string    `json:"name"`
+	Created     time.Time `json:"created"`
+	Registered  bool      `json:"registered,omitempty"`
+	Description string    `json:"description,omitempty"`
+	PrivateKey  string    `json:"privateKey,omitempty"`
+}
+
+func (s Space) NewDelegation(email string) (delegation.Delegation, error) {
+	emailDID, err := didmailto.FromEmail(email)
+	if err != nil {
+		return nil, fmt.Errorf("invalid email address: %w", err)
+	}
+
+	spaceSigner, err := signer.Parse(s.PrivateKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse space private key: %w", err)
+	}
+
+	// Grant all necessary capabilities for the space
+	capabilities := []ucan.Capability[ucan.NoCaveats]{
+		ucan.NewCapability("space/*", s.DID.String(), ucan.NoCaveats{}),
+		ucan.NewCapability("upload/*", s.DID.String(), ucan.NoCaveats{}),
+		ucan.NewCapability("store/*", s.DID.String(), ucan.NoCaveats{}),
+		ucan.NewCapability("index/*", s.DID.String(), ucan.NoCaveats{}),
+		ucan.NewCapability("filecoin/*", s.DID.String(), ucan.NoCaveats{}),
+		ucan.NewCapability("blob/*", s.DID.String(), ucan.NoCaveats{}),
```

Comment:
Hmm, I thought we delegated `*` to the email?

If not, can we please add `assert/*` here like https://github.com/storacha/guppy/pull/141

Reply:
Humm seems like a bug then. fwiw this code was moved from https://github.com/storacha/guppy/blob/main/cmd/space.go#L439

so sounds like this should instead mimic/reference the capabilities as they are defined in your link: https://github.com/storacha/guppy/blob/a3a82d77e28ef1892f5ab9864a176764c19d040f/pkg/client/requestaccess.go#L13 ?

Reply:
@Peeja just checking if this was intentional?